bundles/org.openhab.binding.sony/src/main/java/org/openhab/binding/sony/internal/scalarweb/protocols/ScalarWebAppControlProtocol.java

```diff
@@ -0,0 +1,512 @@
+        }
+    }
+
+    /**
+     * Refresh app status for the given URI
+     *
+     * @param channelId the non-null, non-empty channel ID
+     * @param appUri the non-null, non-empty application URI
+     */
+    private void refreshAppStatus(final String channelId, final String appUri) {
+        Validate.notEmpty(channelId, "channelId cannot be empty");
+        Validate.notEmpty(appUri, "appUri cannot be empty");
+        final WebAppStatus webAppStatus = getWebAppStatus();
+        callback.stateChanged(channelId,
+                webAppStatus != null && webAppStatus.isActive()
+                        && StringUtils.equalsIgnoreCase(appUri, webAppStatus.getUrl()) ? SonyUtil.newStringType(START)
+                                : SonyUtil.newStringType(STOP));
+    }
+
+    /**
+     * Refresh status for the status name
+     *
+     * @param channelId the non-null, non-empty channel ID
+     * @param statusName the non-null, non-empty status name
+     */
+    private void refreshStatus(final String channelId, final String statusName) {
+        Validate.notEmpty(channelId, "channelId cannot be empty");
+        Validate.notEmpty(statusName, "statusName cannot be empty");
+        try {
+            final List<ApplicationStatusList> statuses = execute(ScalarWebMethod.GETAPPLICATIONSTATUSLIST)
+                    .asArray(ApplicationStatusList.class);
+            for (final ApplicationStatusList status : statuses) {
+                if (StringUtils.equalsIgnoreCase(statusName, status.getName())) {
+                    callback.stateChanged(channelId, status.isOn() ? OnOffType.ON : OnOffType.OFF);
+                }
+            }
+            callback.stateChanged(channelId, OnOffType.OFF);
+        } catch (final IOException e) {
+            logger.debug("Exception getting application status list: {}", e.getMessage());
+        }
+
+    }
+
+    /**
+     * Refresh text from a text form
+     *
+     * @param channelId the non-null, non-empty channel ID
+     */
+    private void refreshTextForm(final String channelId) {
+        Validate.notEmpty(channelId, "channelId cannot be empty");
+        try {
+            // Alwasy assume non encrypted version
+            // final String localEncKey = pubKey;
+            // if (localEncKey != null && StringUtils.isNotEmpty(localEncKey)) {
+                final TextFormResult form = execute(ScalarWebMethod.GETTEXTFORM, version -> {
+                    // if (VersionUtilities.equals(version, ScalarWebMethod.V1_0)) {
+                        return null;
+                    // }
+                    // return new TextFormRequest_1_1(rsa encrypted aes key, null);
+                }).as(TextFormResult.class);
+                callback.stateChanged(channelId, SonyUtil.newStringType(form.getText()));
+            // }
+        } catch (final IOException e) {
+            logger.debug("Exception getting text form: {}", e.getMessage());
```

Comment:
would it be worth including the entire exception in the log?

Reply:
I've found it's not worth including the full exception in most of the cases - I use to log the exception everywhere but have really backed off to the spots that I need it.  Most of the time you just need the message to know what's going on.